# OpenDartReader - Users Guide

<img width="40%" src="https://i.imgur.com/FMsL0id.png" >

`OpenDartReader`는 금융감독원 전자공시 시스템의 "Open DART"서비스 API를 손쉽게 사용할 수 있도록 돕는 오픈소스 라이브러리 입니다.

***2020-2022  [FinanceData.KR]() | [facebook.com/financedata]()***

##  OpenDartReader
`Open DART`는 금융감독원이 제공하는 전자공시 시스템에서 제공하는 API서비스 입니다. 기존의 "오픈API", "공시정보 활용마당" 서비스 확대개편하였으며 2020-01-21 (시범) 서비스 시작하였습니다.

`Open DART` API는 매우 잘 만들어진 서비스입니다. 하지만, API를 직접사용하면 추가적인 작업들이 필요합니다. 예를 들어, `Open DART` API는 기업에 대한 내용을 조회할 때, 전자공시에서 사용하는 개별 기업의 고유 번호를 사용합니다. 특정 회사를 상장회사 종목코드으로 조회하려면 매번 고유번호를 얻어야 합니다. 수신한 데이터는 JSON혹은 XML형태 인데 대부분의 경우 pandas 데이터프레임(DataFrame)으로 가공해야 사용하기 더 편리합니다.

`OpenDartReader`는 `Open DART`를 이런 내용을 일반화하고 좀 더 쉽게 `Open DART`를 사용하기 위한 파이썬 라이브러리 입니다. 

## Installation

다음과 같이 설치 합니다.

```bash
pip install opendartreader
```

이미 설치되어 있고 업그레이드가 필요하다면 다음과 같이 설치합니다.
```bash
pip install --upgrade opendartreader 
```



## Quick Start


## 0. OpenDartReader 객체 생성

API키는 전자공시 [인증키 신청 페이지](https://opendart.fss.or.kr/uss/umt/EgovMberInsertView.do)에서 회원가입하고 신청하면 바로 발급 받을 수 있습니다. API키는 40자의 영문과 숫자로 구성되어 있습니다.


```python
import OpenDartReader

api_key = 'd81e78ac719d1c1e4ec4558ef22a737ab6cbb4c8'
dart = OpenDartReader(api_key)
```


### 1. 공시정보 ###
```python
# 특정기업(삼성전자) 상장이후 모든 공시 목록 (5,600 건+)
dart.list('삼성전자') # 기업이름 혹은,
dart.list('005930') # 종목코드를 사용할 수 있습니다.

# 특정기업(삼성전자) 특정 날짜 이후 공시목록 (날짜에 다양한 포맷이 가능합니다 2022, '2022-01-01', '20220101' )
dart.list('삼성전자', start='2022-01-01') # 2022-01-01 ~ 오늘
dart.list('005930', start='2022-01-01') # 2022-01-01 ~ 오늘

# 특정기업(삼성전자) 특정 일자 범위(start~end)의 공시목록 (날짜에 다양한 포맷이 가능합니다)
dart.list('005930', start='2022-04-28', end='2022-04-28')

# 특정기업(삼성전자) 1999년~이후 모든 정기보고서 (정정된 공시포함)
dart.list('005930', start='1999-01-01', kind='A', final=False)

# 특정기업(삼성전자) 1999년~이후 모든 정기보고서 (최종보고서)
dart.list('005930', start='1999-01-01', kind='A') 

# 2022-07-01 하루동안 모든 기업의 공시목록
dart.list(start='20200701', end='20200701')

# 2022-01-01 ~ 2022-01-10 모든 회사의 모든 공시목록 (3,139 건)
dart.list(start='2022-01-01', end='2022-01-10')

# 2022-01-01 ~ 2022-01-10 모든 회사의 모든 공시목록 (정정된 공시포함) (3,587 건)
dart.list(start='2022-01-01', end='2022-01-10', final=False)

# 2022-01-01~2022-03-30 모든 회사의 정기보고서 (corp를 특정 하지 않으면 최대 3개월) (2,352 건)
dart.list(start='2022-01-01', end='2022-03-30', kind='A')


# ==== 1-2. 공시정보 - 기업개황 ====
# 기업의 개황정보
dart.company('005930')

# 회사명에 "삼성전자"가 포함된 회사들에 대한 개황정보
dart.company_by_name('삼성전자')

# ==== 1-2. 공시정보 - 기업개황 ====
# 기업의 개황정보
dart.company('005930')

# 회사명에 "삼성전자"가 포함된 회사들에 대한 개황정보
dart.company_by_name('삼성전자')


# ==== 1-3. 공시정보 - 공시서류원본파일 ====
# 삼성전자 사업보고서 (2022년 반기사업보고서) 원문 텍스트
xml_text = dart.document('20220816001711')


# ==== 1-4. 공시정보 - 고유번호 ====
# 종목코드로 고유번호 얻기
dart.find_corp_code('005930')

# 기업명으로 고유번호 얻기
dart.find_corp_code('삼성전자')

```


### 2. 사업보고서 ###
조회가능한 사업보고서의 항목: ['조건부자본증권미상환', '미등기임원보수', '회사채미상환', '단기사채미상환', '기업어음미상환', '채무증권발행', '사모자금사용', '공모자금사용', '임원전체보수승인', '임원전체보수유형', '주식총수', '회계감사', '감사용역', '회계감사용역계약', '사외이사', '신종자본증권미상환', '증자', '배당', '자기주식', '최대주주', '최대주주변동', '소액주주', '임원', '직원', '임원개인보수', '임원전체보수', '개인별보수', '타법인출자']

```python
dart.report('005930', '미등기임원보수', 2021)  # 미등기임원 보수현황
dart.report('005930', '증자', 2021) # 증자(감자) 현황
dart.report('005930', '배당', 2018)  # 배당에 관한 사항
```

### 3. 상장기업 재무정보 ###
상장법인(금융업 제외)의 주요계정과목(재무상태표, 손익계산서)

```python
# 삼성전자 2021 재무제표 
dart.finstate('삼성전자', 2021) 

# 삼성전자 2021Q1 재무제표
dart.finstate('삼성전자', 2021, reprt_code='11013')

# 여러 상장법인(금융업 제외)의 주요계정과목(재무상태표, 손익계산서)
dart.finstate('00126380,00164779,00164742', 2021)
dart.finstate('005930, 000660, 005380', 2021)
dart.finstate('삼성전자, SK하이닉스, 현대자동차', 2021)

# 재무제표 XBRL 원본 파일 저장 (삼성전자 2021 사업보고서)
dart.finstate_xml('20220308000798', save_as='삼성전자_2021_사업보고서_XBRL.zip')

# 전체 재무제표 (삼성전자 2021 전체 재무제표)
dart.finstate_all('005930', 2021)

# XBRL 표준계정과목체계(계정과목)
dart.xbrl_taxonomy('BS1')
```

### 4. 지분공시 ###
```python
# 대량보유 상황보고 (종목코드, 종목명, 고유번호 모두 지정 가능)
dart.major_shareholders('삼성전자')

# 임원ㆍ주요주주 소유보고 (종목코드, 종목명, 고유번호 모두 지정 가능)
dart.major_shareholders_exec('005930')
```

### 5. 주요사항보고서  ###
```dart.event(corp, event, start=None, end=None)```

주요사항 조회가능 항목: ['부도발생', '영업정지', '회생절차', '해산사유', '유상증자', '무상증자', '유무상증자', '감자', '관리절차개시', '소송', '해외상장결정', '해외상장폐지결정', '해외상장', '해외상장폐지', '전환사채발행', '신주인수권부사채발행', '교환사채발행', '관리절차중단', '조건부자본증권발행', '자산양수도', '타법인증권양도', '유형자산양도', '유형자산양수', '타법인증권양수', '영업양도', '영업양수', '자기주식취득신탁계약해지', '자기주식취득신탁계약체결', '자기주식처분', '자기주식취득', '주식교환', '회사분할합병', '회사분할', '회사합병', '사채권양수', '사채권양도결정']

```python
dart.event('052220', '영업정지', '2019') # iMBC(052220)
dart.event('라이트론', '회생절차', '2019') # 라이트론(069540)
dart.event('휴림네트웍스', '유상증자') # 휴림네트웍스(192410)
dart.event('미원상사', '무상증자') # 미원상사(084990)
dart.event('헬릭스미스', '유무상증자') # 헬릭스미스(084990)
dart.event('미원상사', '감자') # 미원상사(084990)
dart.event('한국조선해양', '소송', '2019') # 한국조선해양(009540)
dart.event('035720', '해외상장결정', '2017') # 카카오(035720)
dart.event('017810', '전환사채발행') # 풀무원(017810)
dart.event('키다리스튜디오', '신주인수권부사채발행') #  키다리스튜디오(020120) 
dart.event('이스트소프트', '교환사채발행') # 이스트소프트(047560) 
```

### 6. 증권신고서   ###
```dart.regstate(corp, key_word, start=None, end=None)```

조회가능한 증권신고서 항목: ['주식의포괄적교환이전', '합병', '증권예탁증권', '채무증권', '지분증권', '분할']
종목코드와 종목명 모두 사용할 수 있습니다. 기간(start, end)를 지정하지 않으면 기본값은 1900~현재 입니다

```python
dart.regstate('하림지주', '주식의포괄적교환이전')
dart.regstate('사조대림', '합병')
dart.regstate('에스앤케이', '증권예탁증권')
dart.regstate('BNK금융지주', '채무증권')
dart.regstate('금호전기', '지분증권')
dart.regstate('케이씨씨', '분할')
```

### 7. 확장 기능   ###
```python
# 지정한 날짜의 공시목록 전체 (공시 시간 정보 포함, 종목코드 미포함)
dart.list_date_ex('2022-10-03')

# 하위 문서(sub_docs), 첨부 문서(attach_docs)
rcp_no = '20220308000798' 
dart.sub_docs(rcp_no) # 하위 문서 제목과 URL
dart.sub_docs(rcp_no, match='감사의견')  # 제목이 잘 매치되는 순서로 소트
dart.attach_docs(rcp_no) # 첨부 문서(i.e. 감사보고서) 제목과 URL
dart.attach_docs(rcp_no, match='감사의의견서') # 제목이 잘 매치되는 순서로 소트

# 첨부 파일 제목과 URL
dart.attach_files(rcp_no)

# URL을 지정한 파일로 다운로드
url = 'http://dart.fss.or.kr/pdf/download/excel.do?rcp_no=20220308000798&dcm_no=8446647&lang=ko'
fn = '[삼성전자]사업보고서_재무제표(2022.03.08)_ko.xls'
dart.download(url, fn)
```

### 0. OpenDartReader 객체 생성 ###

가장 먼저 해야할 일은 발급 받은 API키를 사용하여 `OpenDartReader` 객체를 생성하는 것입니다. 

API키는 전자공시 [인증키 신청 페이지](https://opendart.fss.or.kr/uss/umt/EgovMberInsertView.do)에서 회원가입하고 신청하면 바로 발급 받을 수 있습니다. API키는 40자의 영문과 숫자로 구성되어 있습니다.


In [1]:
import OpenDartReader

api_key = 'c19d19f2fae8006979b6e3670c8e8afe9d7d44cf'

dart = OpenDartReader(api_key)

### 1. 공시정보 ###

#### list() - 공시검색  (개별종목)

지정한 회사의 보고서를 검색합니다. 기간과 보고서의 종류를 지정할 수 있습니다.

```python
dart.list(corp, start=None, end=None, kind='', kind_detail='', final=True)
```

#### 날짜 지정하는 방법
`start`와 `end`에 지정하는 날짜의 형식 '2020-07-01', '2020-7-1', '20200701', '1 july 2020', 'JULY 1 2020' 모두 가능합니다. datetime객체도 가능합니다.

* start 와 end를 함께 지정하면 start~end 기간을 지정합니다.
* start만 지정하면 start 부터 현재까지,
* end만 지정하면 end 하루를 지정하게 됩니다.

In [2]:
# 삼성전자 상장이후 모든 공시 목록 (5,600 건+)
dart.list('삼성전자')

corp_code corp_name stock_code corp_cls              report_nm  \
0     00126380      삼성전자     005930        Y    임원ㆍ주요주주특정증권등소유상황보고서   
1     00126380      삼성전자     005930        Y    임원ㆍ주요주주특정증권등소유상황보고서   
2     00126380      삼성전자     005930        Y    임원ㆍ주요주주특정증권등소유상황보고서   
3     00126380      삼성전자     005930        Y         최대주주등소유주식변동신고서   
4     00126380      삼성전자     005930        Y      기업설명회(IR)개최(안내공시)   
...        ...       ...        ...      ...                    ...   
5624  00126380      삼성전자     005930        Y           채무보증(담보제공포함)   
5625  00126380      삼성전자     005930        Y          최대주주등을위한금전의대여   
5626  00126380      삼성전자     005930        Y        감사보고서 (1999.12)   
5627  00126380      삼성전자     005930        Y  [기재정정]반기보고서 (1999.06)   
5628  00126380      삼성전자     005930        Y  [기재정정]사업보고서 (1998.12)   

            rcept_no  flr_nm  rcept_dt rm  
0     20221018000007     이동하  20221018     
1     20221018000004     박훈철  20221018     
2     20221017000310     박태훈  20221017     
3     20221011800602    삼성전자  20221011  유  
4     20221007800064    삼성전자  20221007  유  
...              ...     ...       ... ..  
5624  20000404000429    삼성전자  20000404     
5625  20000401000031    삼성전자  20000401     
5626  20000320000124  삼일회계법인  20000320     
5627  19990824000003    삼성전자  19990824     
5628  19990626000007    삼성전자  19990626  연  

[5629 rows x 9 columns]

In [3]:
# 특정기업(삼성전자) 특정 일자 범위(start~end)의 공시목록 (날짜에 다양한 포맷이 가능합니다)
dart.list('005930', start='2022-04-28', end='2022-04-28')

corp_code corp_name stock_code corp_cls                     report_nm  \
0   00126380      삼성전자     005930        Y                  기타경영사항(자율공시)   
1   00126380      삼성전자     005930        Y                 특수관계인에대한부동산임대   
2   00126380      삼성전자     005930        Y                 특수관계인에대한부동산임대   
3   00126380      삼성전자     005930        Y                    특수관계인에대한증여   
4   00126380      삼성전자     005930        Y                    특수관계인에대한증여   
5   00126380      삼성전자     005930        Y                    특수관계인에대한증여   
6   00126380      삼성전자     005930        Y                   특수관계인과의보험거래   
7   00126380      삼성전자     005930        Y                  특수관계인에대한담보제공   
8   00126380      삼성전자     005930        Y                  특수관계인에대한담보제공   
9   00126380      삼성전자     005930        Y                  특수관계인에대한담보제공   
10  00126380      삼성전자     005930        Y                 특수관계인에대한부동산임대   
11  00126380      삼성전자     005930        Y                 특수관계인에대한부동산임대   
12  00126380      삼성전자     005930        Y                 특수관계인에대한부동산임대   
13  00126380      삼성전자     005930        Y           임원ㆍ주요주주특정증권등소유상황보고서   
14  00126380      삼성전자     005930        Y                     현금ㆍ현물배당결정   
15  00126380      삼성전자     005930        Y        연결재무제표기준영업(잠정)실적(공정공시)   
16  00126380      삼성전자     005930        Y  [기재정정]연결재무제표기준영업(잠정)실적(공정공시)   

          rcept_no flr_nm  rcept_dt rm  
0   20220428800750   삼성전자  20220428  유  
1   20220428000755   삼성전자  20220428  공  
2   20220428000752   삼성전자  20220428  공  
3   20220428000739   삼성전자  20220428  공  
4   20220428000734   삼성전자  20220428  공  
5   20220428000731   삼성전자  20220428  공  
6   20220428000726   삼성전자  20220428  공  
7   20220428000720   삼성전자  20220428  공  
8   20220428000717   삼성전자  20220428  공  
9   20220428000716   삼성전자  20220428  공  
10  20220428000710   삼성전자  20220428  공  
11  20220428000703   삼성전자  20220428  공  
12  20220428000697   삼성전자  20220428  공  
13  20220428000592    이정배  20220428     
14  20220428800081   삼성전자  20220428  유  
15  20220428800063   삼성전자  20220428  유  
16  20220428800061   삼성전자  20220428  유

In [4]:
# 특정기업(삼성전자) 1999년~이후 모든 정기보고서 (정정된 공시포함)
dart.list('005930', start='1999-01-01', kind='A', final=False)

corp_code corp_name stock_code corp_cls              report_nm  \
0   00126380      삼성전자     005930        Y        반기보고서 (2022.06)   
1   00126380      삼성전자     005930        Y        분기보고서 (2022.03)   
2   00126380      삼성전자     005930        Y        사업보고서 (2021.12)   
3   00126380      삼성전자     005930        Y        분기보고서 (2021.09)   
4   00126380      삼성전자     005930        Y        반기보고서 (2021.06)   
..       ...       ...        ...      ...                    ...   
93  00126380      삼성전자     005930        Y        분기보고서 (2000.03)   
94  00126380      삼성전자     005930        Y  [기재정정]반기보고서 (1999.06)   
95  00126380      삼성전자     005930        Y        반기보고서 (1999.06)   
96  00126380      삼성전자     005930        Y  [기재정정]사업보고서 (1998.12)   
97  00126380      삼성전자     005930        Y        사업보고서 (1998.12)   

          rcept_no flr_nm  rcept_dt  rm  
0   20220816001711   삼성전자  20220816      
1   20220516001751   삼성전자  20220516      
2   20220308000798   삼성전자  20220308   연  
3   20211115001965   삼성전자  20211115      
4   20210817001416   삼성전자  20210817      
..             ...    ...       ...  ..  
93  20000515000739   삼성전자  20000515      
94  19990824000003   삼성전자  19990824      
95  19990814000182   삼성전자  19990814   정  
96  19990626000007   삼성전자  19990626   연  
97  19990403000009   삼성전자  19990403  정연  

[98 rows x 9 columns]

In [5]:
# 특정기업(삼성전자) 1999년~이후 모든 정기보고서 (최종보고서)
dart.list('005930', start='1999-01-01', kind='A') 

corp_code corp_name stock_code corp_cls              report_nm  \
0   00126380      삼성전자     005930        Y        반기보고서 (2022.06)   
1   00126380      삼성전자     005930        Y        분기보고서 (2022.03)   
2   00126380      삼성전자     005930        Y        사업보고서 (2021.12)   
3   00126380      삼성전자     005930        Y        분기보고서 (2021.09)   
4   00126380      삼성전자     005930        Y        반기보고서 (2021.06)   
..       ...       ...        ...      ...                    ...   
88  00126380      삼성전자     005930        Y        반기보고서 (2000.06)   
89  00126380      삼성전자     005930        Y  [첨부추가]사업보고서 (1999.12)   
90  00126380      삼성전자     005930        Y        분기보고서 (2000.03)   
91  00126380      삼성전자     005930        Y  [기재정정]반기보고서 (1999.06)   
92  00126380      삼성전자     005930        Y  [기재정정]사업보고서 (1998.12)   

          rcept_no flr_nm  rcept_dt rm  
0   20220816001711   삼성전자  20220816     
1   20220516001751   삼성전자  20220516     
2   20220308000798   삼성전자  20220308  연  
3   20211115001965   삼성전자  20211115     
4   20210817001416   삼성전자  20210817     
..             ...    ...       ... ..  
88  20000814000482   삼성전자  20000814     
89  20000330000796   삼성전자  20000330  연  
90  20000515000739   삼성전자  20000515     
91  19990824000003   삼성전자  19990824     
92  19990626000007   삼성전자  19990626  연  

[93 rows x 9 columns]

#### list() - 공시검색 (모든종목)
회사(종목)를 지정하지 않으면 모든 종목에 대한 공시를 검색합니다. (이때, 검색 기간은 `start` 와 `end` 사이의 3개월 이내 범위만 가능합니다). 주로 모든 회사를 대상으로 최근 공시를 조회할 목적으로 사용합니다

In [6]:
# 2022-07-01 하루동안 모든 기업의 공시목록
dart.list(start='20200701', end='20200701')

corp_code   corp_name stock_code corp_cls  \
0    00260569         유니셈     036200        K   
1    00373784     스팍스자산운용                   E   
2    00260453      삼성자산운용                   E   
3    00104500      KB자산운용                   E   
4    00104500      KB자산운용                   E   
..        ...         ...        ...      ...   
479  00326272  하나유비에스자산운용                   E   
480  00326272  하나유비에스자산운용                   E   
481  00326272  하나유비에스자산운용                   E   
482  00326272  하나유비에스자산운용                   E   
483  00326272  하나유비에스자산운용                   E   

                                             report_nm        rcept_no  \
0                              [첨부추가]단일판매ㆍ공급계약체결(자율공시)  20200701900516   
1            증권발행실적보고서(집합투자증권)(스팍스본재팬연금저축증권자투자신탁H[주식])  20200701000032   
2    [기재정정]증권신고서(집합투자증권-신탁형)(삼성지수연계증권투자신탁11호[ELS-파생형])  20200701000453   
3    [기재정정]증권신고서(집합투자증권-신탁형)(KB파워리자드지수연계증권투자신탁SEK-5...  20200701000391   
4    [기재정정]증권신고서(집합투자증권-신탁형)(KB지수연계더블리자드증권투자신탁57호(E...  20200701000378   
..                                                 ...             ...   
479          효력발생안내( 2020.6.25. 제출 증권신고서(집합투자증권-신탁형) )  20200630100014   
480          효력발생안내( 2020.6.25. 제출 일괄신고서(집합투자증권-신탁형) )  20200630100013   
481          효력발생안내( 2020.6.25. 제출 증권신고서(집합투자증권-신탁형) )  20200630100012   
482          효력발생안내( 2020.6.25. 제출 증권신고서(집합투자증권-신탁형) )  20200630100011   
483          효력발생안내( 2020.6.25. 제출 증권신고서(집합투자증권-신탁형) )  20200630100010   

      flr_nm  rcept_dt rm  
0        유니셈  20200701  코  
1    스팍스자산운용  20200701     
2     삼성자산운용  20200701     
3     KB자산운용  20200701     
4     KB자산운용  20200701     
..       ...       ... ..  
479    금융감독원  20200701     
480    금융감독원  20200701     
481    금융감독원  20200701     
482    금융감독원  20200701     
483    금융감독원  20200701     

[484 rows x 9 columns]

In [7]:
# 2022-01-01 ~ 2022-01-10 모든 회사의 모든 공시목록 (3,139 건)
dart.list(start='2022-01-01', end='2022-01-10')

corp_code              corp_name stock_code corp_cls  \
0     01578067      뉴스타해피제삼차유동화전문유한회사                   E   
1     00498162               한국주택금융공사                   E   
2     00419484           현대인베스트먼트자산운용                   E   
3     00259776               미래에셋자산운용                   E   
4     01606351  도이치파이낸셜오토제오차유동화전문유한회사                   E   
...        ...                    ...        ...      ...   
3128  00324548               한국투자신탁운용                   E   
3129  00324548               한국투자신탁운용                   E   
3130  00860794          디비씨앤에스자동차손해사정                   E   
3131  00860794          디비씨앤에스자동차손해사정                   E   
3132  00860794          디비씨앤에스자동차손해사정                   E   

                                              report_nm        rcept_no  \
0                                           자산양도등의등록신청서  20220110000065   
1                                 채권양도등의등록신청서[한국주택금융공사]  20220106000367   
2         일괄신고서(집합투자증권-신탁형)(현대인베스트먼트코리아ESG증권투자신탁1호(채권))  20220105000105   
3     일괄신고서(집합투자증권-신탁형)(미래에셋TIGERKEDI혁신기업ESG30증권상장지수...  20220104000319   
4                                자산유동화계획의등록신청서(유동화전문회사)  20220105000521   
...                                                 ...             ...   
3128         효력발생안내( 2021.12.30. 제출 일괄신고서(집합투자증권-신탁형) )  20211231100015   
3129         효력발생안내( 2021.12.30. 제출 일괄신고서(집합투자증권-신탁형) )  20211231100014   
3130                                     특수관계인으로부터부동산임차  20211231000498   
3131                                     특수관계인으로부터부동산임차  20211231000497   
3132                                        특수관계인과의내부거래  20211231000496   

                     flr_nm  rcept_dt rm  
0                        다날  20220110     
1                  한국주택금융공사  20220106     
2              현대인베스트먼트자산운용  20220105     
3                  미래에셋자산운용  20220104     
4     도이치파이낸셜오토제오차유동화전문유한회사  20220105     
...                     ...       ... ..  
3128                  금융감독원  20220103     
3129                  금융감독원  20220103     
3130          디비씨앤에스자동차손해사정  20220103  공  
3131          디비씨앤에스자동차손해사정  20220103  공  
3132          디비씨앤에스자동차손해사정  20220103  공  

[3133 rows x 9 columns]

In [8]:
# 2022-01-01 ~ 2022-01-10 모든 회사의 모든 공시목록 (정정된 공시포함) (3,587 건)
dart.list(start='2022-01-01', end='2022-01-10', final=False)

corp_code      corp_name stock_code corp_cls  \
0     00498162       한국주택금융공사                   E   
1     00882635         하우자산운용                   E   
2     00760494       얼라이언스번스틴                   E   
3     00686776        BNK자산운용                   E   
4     00267331         다올자산운용                   E   
...        ...            ...        ...      ...   
3582  00324548       한국투자신탁운용                   E   
3583  00324548       한국투자신탁운용                   E   
3584  00860794  디비씨앤에스자동차손해사정                   E   
3585  00860794  디비씨앤에스자동차손해사정                   E   
3586  00860794  디비씨앤에스자동차손해사정                   E   

                                              report_nm        rcept_no  \
0                               채권유동화계획의등록신청서[한국주택금융공사]  20220106000410   
1         [기재정정]투자설명서(집합투자증권)(하우플러스타리츠부동산투자신탁(리츠-재간접형))  20220110000267   
2                       [기재정정]일괄신고서(집합투자증권-신탁형)(ABFCPI)  20220105000461   
3     증권신고서(집합투자증권-신탁형)(BNK더블리자드지수연계증권투자신탁KSE-35호[EL...  20220107000516   
4     증권신고서(집합투자증권-신탁형)(다올KTB지수연계플러스증권투자신탁NSE-97호[EL...  20220107000281   
...                                                 ...             ...   
3582         효력발생안내( 2021.12.30. 제출 일괄신고서(집합투자증권-신탁형) )  20211231100015   
3583         효력발생안내( 2021.12.30. 제출 일괄신고서(집합투자증권-신탁형) )  20211231100014   
3584                                     특수관계인으로부터부동산임차  20211231000498   
3585                                     특수관계인으로부터부동산임차  20211231000497   
3586                                        특수관계인과의내부거래  20211231000496   

             flr_nm  rcept_dt rm  
0          한국주택금융공사  20220106  정  
1            하우자산운용  20220110  정  
2          얼라이언스번스틴  20220105  정  
3           BNK자산운용  20220107  정  
4            다올자산운용  20220107  정  
...             ...       ... ..  
3582          금융감독원  20220103     
3583          금융감독원  20220103     
3584  디비씨앤에스자동차손해사정  20220103  공  
3585  디비씨앤에스자동차손해사정  20220103  공  
3586  디비씨앤에스자동차손해사정  20220103  공  

[3587 rows x 9 columns]

In [9]:
# 2021-01-01 부터 현재까지 모든 회사의 정기보고서
dart.list(start='2021-01-01', kind='A')

{"status":"100","message":"corp_code가 없는 경우 검색기간은 3개월만 가능합니다."}


Empty DataFrame
Columns: []
Index: []

In [10]:
# 2022-01-01~2022-03-30 모든 회사의 정기보고서 (corp를 특정 하지 않으면 최대 3개월) (2,352 건)
dart.list(start='2022-01-01', end='2022-03-30', kind='A')

corp_code  corp_name stock_code corp_cls              report_nm  \
0     01309209  더블다운인터액티브                   E  [첨부추가]사업보고서 (2021.12)   
1     01011562      우성아이비     194610        E  [첨부추가]사업보고서 (2021.12)   
2     00463449      더존홀딩스                   E  [첨부추가]사업보고서 (2021.12)   
3     00362885     에스텍시스템                   E  [첨부추가]사업보고서 (2021.12)   
4     00259934        예림당     036000        K        사업보고서 (2021.12)   
...        ...        ...        ...      ...                    ...   
2333  00209443    우리금융캐피탈     033660        E  [기재정정]사업보고서 (2020.12)   
2334  00201432      비츠로시스     054220        K  [기재정정]반기보고서 (2021.09)   
2335  00171636      한솔홀딩스     004150        Y  [기재정정]분기보고서 (2021.09)   
2336  00109781       대림제지     017650        K  [기재정정]사업보고서 (2019.12)   
2337  00109781       대림제지     017650        K  [기재정정]사업보고서 (2016.12)   

            rcept_no     flr_nm  rcept_dt rm  
0     20220330000909  더블다운인터액티브  20220330  연  
1     20220328000568      우성아이비  20220328  연  
2     20220328000811      더존홀딩스  20220328  연  
3     20220330000029     에스텍시스템  20220330  연  
4     20220330001786        예림당  20220330  연  
...              ...        ...       ... ..  
2333  20220111000355    우리금융캐피탈  20220111  연  
2334  20220106000470      비츠로시스  20220106     
2335  20220106000434      한솔홀딩스  20220106     
2336  20220103000285       대림제지  20220103     
2337  20220103000219       대림제지  20220103     

[2338 rows x 9 columns]

#### company(), company_by_name() - 기업개황

```python
# 기업의 개황정보를 읽어옵니다
dart.company(corp) 

# 기업을 검색하여 이름이 포함된 모든 기업들의 기업개황 정보를 반환합니다.
dart.company_by_name(name)
```

In [11]:
dart.company('005930') # 삼성전자, 005930, 00126380

{'status': '000',
 'message': '정상',
 'corp_code': '00126380',
 'corp_name': '삼성전자(주)',
 'corp_name_eng': 'SAMSUNG ELECTRONICS CO,.LTD',
 'stock_name': '삼성전자',
 'stock_code': '005930',
 'ceo_nm': '한종희, 경계현',
 'corp_cls': 'Y',
 'jurir_no': '1301110006246',
 'bizr_no': '1248100998',
 'adres': '경기도 수원시 영통구  삼성로 129 (매탄동)',
 'hm_url': 'www.sec.co.kr',
 'ir_url': '',
 'phn_no': '031-200-1114',
 'fax_no': '031-200-7538',
 'induty_code': '264',
 'est_dt': '19690113',
 'acc_mt': '12'}

find_company_by_name()는 기업을 검색하여 이름이 포함된 모든 기업들의 기업개황 정보를 반환합니다.

개별 회사의 corp_code(고유번호), stock_code (종목코드)를 얻는데도 유용하게 사용할 수 있습니다.

In [12]:
# 회사명에 삼성전자가 포함된 회사들에 대한 개황정보
data = dart.company_by_name('삼성전자')
data[:2]

[{'status': '000',
  'message': '정상',
  'corp_code': '00366997',
  'corp_name': '삼성전자로지텍(주)',
  'corp_name_eng': 'SELC.CO.,LTD',
  'stock_name': '삼성전자로지텍',
  'stock_code': '',
  'ceo_nm': '최윤범',
  'corp_cls': 'E',
  'jurir_no': '1301110046797',
  'bizr_no': '1248155381',
  'adres': '경기도 수원시 영통구 삼성로 129',
  'hm_url': 'www.selc.co.kr',
  'ir_url': '',
  'phn_no': '031-270-3488',
  'fax_no': '031-270-3673',
  'induty_code': '52999',
  'est_dt': '19980401',
  'acc_mt': '12'},
 {'status': '000',
  'message': '정상',
  'corp_code': '00252074',
  'corp_name': '삼성전자판매 주식회사',
  'corp_name_eng': 'SAMSUNG ELECTRONICS SALES Co., Ltd.',
  'stock_name': '삼성전자판매',
  'stock_code': '',
  'ceo_nm': '이평우',
  'corp_cls': 'E',
  'jurir_no': '1801110210300',
  'bizr_no': '6178117517',
  'adres': '경기도 성남시 분당구 황새울로 340',
  'hm_url': 'www.samsungsales.co.kr',
  'ir_url': '',
  'phn_no': '02-3460-6656',
  'fax_no': '02-3460-6802',
  'induty_code': '47320',
  'est_dt': '19960711',
  'acc_mt': '12'}]

가져온 기업개황 정보(dict list)를 데이터 프레임으로 가공하고, 필요한 컬럼을 지정하여 정리할 수 있습니다.

In [13]:
import pandas as pd

pd.DataFrame(data)[['stock_name' , 'stock_code' , 'ceo_nm' , 'corp_cls' , 'jurir_no' , 'bizr_no' , 'adres' ]]

stock_name stock_code    ceo_nm corp_cls       jurir_no     bizr_no  \
0     삼성전자로지텍                  최윤범        E  1301110046797  1248155381   
1      삼성전자판매                  이평우        E  1801110210300  6178117517   
2  삼성전자서비스씨에스                  정석진        E  1358110352541  7898701103   
3        삼성전자     005930  한종희, 경계현        Y  1301110006246  1248100998   
4     삼성전자서비스                  송봉섭        E  1301110049139  1248158485   

                                 adres  
0                  경기도 수원시 영통구 삼성로 129  
1                 경기도 성남시 분당구 황새울로 340  
2  경기도 수원시 영통구 중부대로 324 (매탄동, KT동수원지사)  
3           경기도 수원시 영통구  삼성로 129 (매탄동)  
4            경기도 수원시 영통구 삼성로 290 (원천동)

#### document() - 공시서류 원문 

In [14]:
# 삼성전자 사업보고서 (2022년 반기사업보고서) 원문 텍스트
# http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20220816001711 
     
xml_text = dart.document('20220816001711') # 2021 사업보고서
xml_text[:2000]

'<?xml version="1.0" encoding="utf-8"?>\r\n\r\n\r\n<DOCUMENT xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="dart3.xsd">\n\n<DOCUMENT-NAME ACODE="11012">반기보고서</DOCUMENT-NAME>\n<FORMULA-VERSION ADATE="20220418">4.8</FORMULA-VERSION>\n<COMPANY-NAME AREGCIK="00126380">삼성전자주식회사</COMPANY-NAME>\n\n<SUMMARY>\n<EXTRACTION ACODE="CRP_RGS_NO_TEMP" AFEATURE="BOTH">130111-0006246</EXTRACTION>\n</SUMMARY>\n\n\n<BODY>\n\n<COVER>\n<P></P>\n<COVER-TITLE ATOC="Y" AASSOCNOTE="COVER">반 기 보 고 서</COVER-TITLE>\n<P></P>\n<P USERMARK="F-16">                                    (제 54 기)</P>\n\n<TABLE-GROUP ACLASS="COVER" ADELETETABLE="N">\n\n<TABLE ACLASS="EXTRACTION" AFIXTABLE="Y" WIDTH="600" BORDER="0">\n\n<COLGROUP>\n<COL WIDTH="209"></COL>\n<COL WIDTH="153"></COL>\n<COL WIDTH="211"></COL>\n</COLGROUP>\n\n<TBODY>\n\n<TR ACOPY="N" ADELETE="N">\n<TD CLASS="NORMAL" ROWSPAN="2" ALIGN="RIGHT" VALIGN="MIDDLE" WIDTH="218" HEIGHT="60" AUPDATECONT="N">사업연도</TD>\n<TU CLASS="NORMAL"

#### find_corp_code() - 고유번호 얻기
종목코드 혹은 기업명으로 고유번호를 가져옵니다.
전자공시에서 개별 기업은 고유번호로 식별 됩니다. 특히, 상장종목이 아닌 경우는 고유번호를 사용해야 합니다.

In [15]:
dart.find_corp_code('005930')

'00126380'

In [16]:
dart.find_corp_code('삼성전자')

'00126380'

#### corp_codes - 고유번호(속성)
고유번호, 종목명, 종목코드 등의 정보를 포함하고 있는 속성값 입니다.  8만 1천여개 기업에 대한 고유번호, 종목명, 종목코드 등의 정보를 포함하고 있습니다.
 

In [17]:
dart.corp_codes

corp_code          corp_name stock_code modify_date
0      00434003                 다코               20170630
1      00434456               일산약품               20170630
2      00430964              굿앤엘에스               20170630
3      00432403               한라판지               20170630
4      00388953  크레디피아제이십오차유동화전문회사               20170630
...         ...                ...        ...         ...
96186  00913768              파인일렉컴               20220228
96187  00110857             대신염직공업               20220228
96188  01364808              우진알디씨               20220228
96189  01497522            철은인터내셔날               20220228
96190  01494136              군자출판사               20220228

[96191 rows x 4 columns]

### 2. 사업보고서 ###
사업보고서의 주요 내용을 조회 합니다.

```python
dart.report(corp, key_word, bsns_year, reprt_code='11011')
```

`key_word`에 다음 항목중 하나를 지정할 수 있습니다.
> '조건부자본증권미상환', '미등기임원보수', '회사채미상환', '단기사채미상환', '기업어음미상환', '채무증권발행', '사모자금사용', '공모자금사용', '임원전체보수승인', '임원전체보수유형', '주식총수', '회계감사', '감사용역', '회계감사용역계약', '사외이사', '신종자본증권미상환', '증자', '배당', '자기주식', '최대주주', '최대주주변동', '소액주주', '임원', '직원', '임원개인보수', '임원전체보수', '개인별보수', '타법인출자' 

bsns_year 에 사업 년도를 지정합니다 (문자열 혹은 정수값)

reprt_code 에는 보고서 코드(문자열)을 지정합니다. 
> '11013'=1분기보고서, '11012'=반기보고서, '11014'=3분기보고서, '11011'=사업보고서

더 상세한 내용은 [OpenDartReader - Reference Manual]을 참고하십시오.

In [18]:
# 삼성전자(005930), 배당관련 사항, 2021년
dart.report('005930', '배당', 2021) 

rcept_no corp_cls corp_code corp_name              se      thstrm  \
0   20220308000798        Y  00126380      삼성전자       주당액면가액(원)         100   
1   20220308000798        Y  00126380      삼성전자  (연결)당기순이익(백만원)  39,243,791   
2   20220308000798        Y  00126380      삼성전자  (별도)당기순이익(백만원)  30,970,954   
3   20220308000798        Y  00126380      삼성전자    (연결)주당순이익(원)       5,777   
4   20220308000798        Y  00126380      삼성전자    현금배당금총액(백만원)   9,809,438   
5   20220308000798        Y  00126380      삼성전자    주식배당금총액(백만원)           -   
6   20220308000798        Y  00126380      삼성전자   (연결)현금배당성향(%)       25.00   
7   20220308000798        Y  00126380      삼성전자      현금배당수익률(%)        1.80   
8   20220308000798        Y  00126380      삼성전자      현금배당수익률(%)        2.00   
9   20220308000798        Y  00126380      삼성전자      주식배당수익률(%)           -   
10  20220308000798        Y  00126380      삼성전자      주식배당수익률(%)           -   
11  20220308000798        Y  00126380      삼성전자     주당 현금배당금(원)       1,444   
12  20220308000798        Y  00126380      삼성전자     주당 현금배당금(원)       1,445   
13  20220308000798        Y  00126380      삼성전자      주당 주식배당(주)           -   
14  20220308000798        Y  00126380      삼성전자      주당 주식배당(주)           -   

        frmtrm        lwfr stock_knd  
0          100         100       NaN  
1   26,090,846  21,505,054       NaN  
2   15,615,018  15,353,323       NaN  
3        3,841       3,166       NaN  
4   20,338,075   9,619,243       NaN  
5            -           -       NaN  
6        78.00       44.70       NaN  
7         4.00        2.60       보통주  
8         4.20        3.10       우선주  
9            -           -       보통주  
10           -           -       우선주  
11       2,994       1,416       보통주  
12       2,995       1,417       우선주  
13           -           -       보통주  
14           -           -       우선주

In [19]:
# 서울반도체(046890), 최대주주 관한 사항, 2021년

dart.report('046890', '최대주주', 2021) 

rcept_no corp_cls corp_code corp_name stock_knd rm   nm relate  \
0  20220322000455        K  00130763     서울반도체       보통주  -  이정훈     본인   
1  20220322000455        K  00130763     서울반도체       보통주  -  이민호  특수관계인   
2  20220322000455        K  00130763     서울반도체       보통주  -  이민규  특수관계인   
3  20220322000455        K  00130763     서울반도체       보통주  -    계    NaN   
4  20220322000455        K  00130763     서울반도체         -  -    계    NaN   

  bsis_posesn_stock_co bsis_posesn_stock_qota_rt trmend_posesn_stock_co  \
0            7,922,067                     13.59              7,922,067   
1            5,076,368                      8.71              4,356,368   
2            5,076,358                      8.71              4,356,358   
3           18,074,793                     31.00             16,634,793   
4                    -                         -                      -   

  trmend_posesn_stock_qota_rt  
0                       13.59  
1                        7.47  
2                        7.47  
3                       28.53  
4                           -

In [20]:
# 서울반도체(046890), 임원 관한 사항, 2021

dart.report('046890', '임원', 2021) 

rcept_no corp_cls corp_code corp_name   nm sexdstn   birth_ym  \
0  20220322000455        K  00130763     서울반도체  이정훈       남  1953년 01월   
1  20220322000455        K  00130763     서울반도체  홍명기       남  1973년 05월   
2  20220322000455        K  00130763     서울반도체  이재성       남  1969년 05월   
3  20220322000455        K  00130763     서울반도체  박형건       남  1960년 04월   
4  20220322000455        K  00130763     서울반도체  이창한       남  1957년 05월   
5  20220322000455        K  00130763     서울반도체  홍준표       남  1967년 12월   

       ofcps rgist_exctv_at fte_at   chrg_job  \
0   대표이사\n사장           사내이사     상근       경영전반   
1  대표이사\n부사장           사내이사     상근       전산전반   
2  재무그룹장\n상무           사내이사     상근       재무전반   
3       사외이사           사외이사    비상근  사외이사/감사위원   
4       사외이사           사외이사    비상근  사외이사/감사위원   
5       사외이사           사외이사    비상근  사외이사/감사위원   

                                         main_career mxmm_shrholdr_relate  \
0                  - 삼신전기(주) 부사장\n- 서울반도체(주) 대표이사(現)                   본인   
1  - 현대정보기술㈜ 현대상선 IT실\n- 대영포장㈜ 정보기술팀\n- ㈜비에스지파트너스...                    -   
2  - 한라홀딩스 주식회사 재무팀\n- 만도헬라일렉트로닉스 주식회사 재무팀\n- 만도 ...                    -   
3                 - 국세청 정보공개심의위원회 위원\n- 우리회계법인 대표(現)                    -   
4                  - 키투웨이공공부문 대표\n- 반도체산업협회 상근부회장(現)                    -   
5            - 울산대학교 성형외과 교수(現)\n- 서울아산병원 성형외과 교수(現)                    -   

  hffc_pd  tenure_end_on  
0  1992년~  2023년 03월 30일  
1  2009년~  2022년 03월 27일  
2  2020년~  2023년 03월 30일  
3  2016년~  2022년 03월 30일  
4  2020년~  2022년 03월 27일  
5  2020년~  2022년 03월 27일

In [21]:
# 삼성전자(005930), 미등기임원 보수현황, 2021년

dart.report('005930', '미등기임원보수', 2021)  # 미등기임원 보수현황

rcept_no corp_cls corp_code corp_name     se rm fyer_salary_totamt  \
0  20220308000798        Y  00126380      삼성전자  미등기임원  -    717,856,000,000   

  jan_salary_am nmpr  
0   790,000,000  933

### 3. 상장기업 재무정보 ###

In [22]:
# 삼성전자 2021 재무제표

df = dart.finstate('삼성전자', 2021) # 사업보고서

print('row count', len(df))
df.head()

row count 26


rcept_no reprt_code bsns_year corp_code stock_code fs_div   fs_nm  \
0  20220308000798      11011      2021  00126380     005930    CFS  연결재무제표   
1  20220308000798      11011      2021  00126380     005930    CFS  연결재무제표   
2  20220308000798      11011      2021  00126380     005930    CFS  연결재무제표   
3  20220308000798      11011      2021  00126380     005930    CFS  연결재무제표   
4  20220308000798      11011      2021  00126380     005930    CFS  연결재무제표   

  sj_div  sj_nm account_nm  ...      thstrm_dt        thstrm_amount frmtrm_nm  \
0     BS  재무상태표       유동자산  ...  2021.12.31 현재  218,163,185,000,000    제 52 기   
1     BS  재무상태표      비유동자산  ...  2021.12.31 현재  208,457,973,000,000    제 52 기   
2     BS  재무상태표       자산총계  ...  2021.12.31 현재  426,621,158,000,000    제 52 기   
3     BS  재무상태표       유동부채  ...  2021.12.31 현재   88,117,133,000,000    제 52 기   
4     BS  재무상태표      비유동부채  ...  2021.12.31 현재   33,604,094,000,000    제 52 기   

       frmtrm_dt        frmtrm_amount bfefrmtrm_nm   bfefrmtrm_dt  \
0  2020.12.31 현재  198,215,579,000,000       제 51 기  2019.12.31 현재   
1  2020.12.31 현재  180,020,139,000,000       제 51 기  2019.12.31 현재   
2  2020.12.31 현재  378,235,718,000,000       제 51 기  2019.12.31 현재   
3  2020.12.31 현재   75,604,351,000,000       제 51 기  2019.12.31 현재   
4  2020.12.31 현재   26,683,351,000,000       제 51 기  2019.12.31 현재   

      bfefrmtrm_amount ord currency  
0  181,385,260,000,000   1      KRW  
1  171,179,237,000,000   3      KRW  
2  352,564,497,000,000   5      KRW  
3   63,782,764,000,000   7      KRW  
4   25,901,312,000,000   9      KRW  

[5 rows x 21 columns]

In [23]:
# 삼성전자 2021Q1 재무제표

df = dart.finstate('삼성전자', 2021, reprt_code='11013') # 2018년 1분기 보고서 재무제표

print('row count', len(df))
df.head()

row count 26


rcept_no reprt_code bsns_year corp_code stock_code fs_div   fs_nm  \
0  20210517001185      11013      2021  00126380     005930    CFS  연결재무제표   
1  20210517001185      11013      2021  00126380     005930    CFS  연결재무제표   
2  20210517001185      11013      2021  00126380     005930    CFS  연결재무제표   
3  20210517001185      11013      2021  00126380     005930    CFS  연결재무제표   
4  20210517001185      11013      2021  00126380     005930    CFS  연결재무제표   

  sj_div  sj_nm account_nm   thstrm_nm      thstrm_dt        thstrm_amount  \
0     BS  재무상태표       유동자산  제 53 기1분기말  2021.03.31 현재  209,155,353,000,000   
1     BS  재무상태표      비유동자산  제 53 기1분기말  2021.03.31 현재  183,670,921,000,000   
2     BS  재무상태표       자산총계  제 53 기1분기말  2021.03.31 현재  392,826,274,000,000   
3     BS  재무상태표       유동부채  제 53 기1분기말  2021.03.31 현재   90,109,462,000,000   
4     BS  재무상태표      비유동부채  제 53 기1분기말  2021.03.31 현재   28,448,221,000,000   

  frmtrm_nm      frmtrm_dt        frmtrm_amount ord currency  \
0   제 52 기말  2020.12.31 현재  198,215,579,000,000   1      KRW   
1   제 52 기말  2020.12.31 현재  180,020,139,000,000   3      KRW   
2   제 52 기말  2020.12.31 현재  378,235,718,000,000   5      KRW   
3   제 52 기말  2020.12.31 현재   75,604,351,000,000   7      KRW   
4   제 52 기말  2020.12.31 현재   26,683,351,000,000   9      KRW   

  thstrm_add_amount frmtrm_add_amount  
0               NaN               NaN  
1               NaN               NaN  
2               NaN               NaN  
3               NaN               NaN  
4               NaN               NaN

In [24]:
# 전체 재무제표 (삼성전자 2021 전체 재무제표)
df = dart.finstate_all('005930', 2021)

print('row count', len(df))
df.head()

reprt_code='11011', fs_div='CFS' (사업보고서, 연결제무제표)'
row count 186


rcept_no reprt_code bsns_year corp_code sj_div  sj_nm  \
0  20220308000798      11011      2021  00126380     BS  재무상태표   
1  20220308000798      11011      2021  00126380     BS  재무상태표   
2  20220308000798      11011      2021  00126380     BS  재무상태표   
3  20220308000798      11011      2021  00126380     BS  재무상태표   
4  20220308000798      11011      2021  00126380     BS  재무상태표   

                                          account_id       account_nm  \
0                            ifrs-full_CurrentAssets             유동자산   
1                   ifrs-full_CashAndCashEquivalents         현금및현금성자산   
2  dart_ShortTermDepositsNotClassifiedAsCashEquiv...           단기금융상품   
3                                       -표준계정코드 미사용-      단기상각후원가금융자산   
4  ifrs-full_CurrentFinancialAssetsAtFairValueThr...  단기당기손익-공정가치금융자산   

  account_detail thstrm_nm    thstrm_amount frmtrm_nm    frmtrm_amount  \
0              -    제 53 기  218163185000000    제 52 기  198215579000000   
1              -    제 53 기   39031415000000    제 52 기   29382578000000   
2              -    제 53 기   81708986000000    제 52 기   92441703000000   
3              -    제 53 기    3369034000000    제 52 기    2757111000000   
4              -    제 53 기      40757000000    제 52 기      71451000000   

  bfefrmtrm_nm bfefrmtrm_amount ord currency thstrm_add_amount  
0       제 51 기  181385260000000   1      KRW               NaN  
1       제 51 기   26885999000000   2      KRW               NaN  
2       제 51 기   76252052000000   3      KRW               NaN  
3       제 51 기    3914216000000   4      KRW               NaN  
4       제 51 기    1727436000000   5      KRW               NaN

### 4. 지분공시 ### 
* major_shareholders() - 대량보유 상황보고
* major_shareholders_exec() - 임원ㆍ주요주주 소유보고


In [25]:
df = dart.major_shareholders('삼성전자') # 종목코드, 종목명, 고유번호 모두 가능

print('row count', len(df))
df.head()

row count 22


rcept_no    rcept_dt corp_code corp_name report_tp  repror  \
0  20210503000505  2021-05-03  00126380      삼성전자        일반    삼성물산   
1  20210504000163  2021-05-04  00126380      삼성전자        약식  국민연금공단   
2  20210507000809  2021-05-07  00126380      삼성전자        일반    삼성물산   
3  20210730000519  2021-07-30  00126380      삼성전자        일반    삼성물산   
4  20210806000286  2021-08-06  00126380      삼성전자        일반    삼성물산   

           stkqy   stkqy_irds  stkrt stkrt_irds   ctr_stkqy ctr_stkrt  \
0  1,263,039,758   -3,582,011  21.16      -0.06  88,577,273      1.48   
1    578,352,347  -59,717,656   9.69      -1.00           -         -   
2  1,263,093,732       53,974  21.16       0.00  88,577,273      1.48   
3  1,262,820,989     -272,743  21.15      -0.01  88,577,273      1.48   
4  1,262,889,978       68,989  21.15       0.00  88,577,273      1.48   

                              report_resn  
0  - 특별관계자 및 보유주식수 변동\n- 보유주식등에 관한 계약의 체결  
1                               단순추가취득/처분  
2                      - 특별관계자 및 보유주식수 변동  
3          - 보유주식수 변동\n- 보유주식등에 관한 계약의 변경  
4                      - 특별관계자 및 보유주식수 변동

In [26]:
df = dart.major_shareholders_exec('삼성전자') # 종목코드, 종목명, 고유번호 모두 가능

print('row count', len(df))
df.head()

row count 259


rcept_no    rcept_dt corp_code corp_name  repror isu_exctv_rgist_at  \
0  20201029000242  2020-10-29  00126380      삼성전자     안규리               등기임원   
1  20201030000605  2020-10-30  00126380      삼성전자     윤성혁              비등기임원   
2  20201103000140  2020-11-03  00126380      삼성전자     문형준              비등기임원   
3  20201104000114  2020-11-04  00126380      삼성전자  국민연금공단                  -   
4  20201109000178  2020-11-09  00126380      삼성전자     안규리               등기임원   

  isu_exctv_ofcps isu_main_shrholdr sp_stock_lmp_cnt sp_stock_lmp_irds_cnt  \
0            사외이사                 -            2,400                   100   
1              상무                 -            3,000                 3,000   
2              상무                 -            1,000                 1,000   
3               -           10%이상주주      658,486,630            -2,570,841   
4            사외이사                 -            2,500                   100   

  sp_stock_lmp_rate sp_stock_lmp_irds_rate  
0              0.00                   0.00  
1              0.00                   0.00  
2              0.00                   0.00  
3              9.69                  -0.04  
4              0.00                   0.00

### 5. 주요사항보고서  ###
dart.event(corp, event, start=None, end=None)

사업보고서내 주요사항보고 항목을 가져옵니다.  
조회가능한 주요사항 항목은 다음과 같습니다.
> '부도발생', '영업정지', '회생절차', '해산사유', '유상증자', '무상증자', '유무상증자', '감자', '관리절차개시', '소송', '해외상장결정', '해외상장폐지결정', '해외상장', '해외상장폐지', '전환사채발행', '신주인수권부사채발행', '교환사채발행', '관리절차중단', '조건부자본증권발행', '자산양수도', '타법인증권양도', '유형자산양도', '유형자산양수', '타법인증권양수', '영업양도', '영업양수', '자기주식취득신탁계약해지', '자기주식취득신탁계약체결', '자기주식처분', '자기주식취득', '주식교환', '회사분할합병', '회사분할', '회사합병', '사채권양수', '사채권양도결정'

In [27]:
dart.event('052220', '영업정지', '2019') # iMBC(052220)
dart.event('라이트론', '회생절차', '2019') # 라이트론(069540)
dart.event('휴림네트웍스', '유상증자') # 휴림네트웍스(192410)
dart.event('미원상사', '무상증자') # 미원상사(084990)
dart.event('헬릭스미스', '유무상증자') # 헬릭스미스(084990)
dart.event('미원상사', '감자') # 미원상사(084990)
dart.event('한국조선해양', '소송', '2019') # 한국조선해양(009540)
dart.event('035720', '해외상장결정', '2017') # 카카오(035720)
dart.event('017810', '전환사채발행') # 풀무원(017810)
dart.event('키다리스튜디오', '신주인수권부사채발행') #  키다리스튜디오(020120) 
dart.event('이스트소프트', '교환사채발행') # 이스트소프트(047560) 

rcept_no corp_cls corp_code corp_name           bddd od_a_at_t  \
0  20191212000301        K  00273420    이스트소프트  2019년 12월 10일         3   

  od_a_at_b adt_a_atn fdpp_fclt fdpp_bsninh  ...  \
0         -        참석         -           -  ...   

                                ex_sm_r ovis_ltdtl ex_rt ex_prc  \
0  사모발행으로 증권신고서 제출면제 (발행 후 1년간 권면분할 금지)          -   100  5,080   

                                         ex_prc_dmth            extg  \
0  교환대상 주식1주로 교환되는 가격(이하 "교환가격”이라 함)은 본 사채 발행을 위한...  줌인터넷 주식회사 보통주식   

  extg_stkcnt extg_tisstk_vs     exrqpd_bgd     exrqpd_edd  
0   1,181,102           4.60  2019년 12월 13일  2022년 12월 05일  

[1 rows x 42 columns]


### 6. 증권신고서   ###

In [28]:
# dart.regstate(corp, key_word, start=None, end=None)
# 조회가능한 증권신고서 항목: ['주식의포괄적교환이전', '합병', '증권예탁증권', '채무증권', '지분증권', '분할']

dart.regstate('하림지주', '주식의포괄적교환이전')
dart.regstate('사조대림', '합병')
dart.regstate('에스앤케이', '증권예탁증권')
dart.regstate('BNK금융지주', '채무증권')
dart.regstate('금호전기', '지분증권')
dart.regstate('케이씨씨', '분할')

rcept_no corp_cls corp_code corp_name        rpt_rcpn  stn  \
0  20191015000270        Y  00105271      케이씨씨  20191008000558   분할   
0  20191015000270        Y  00105271      케이씨씨             NaN  NaN   
0  20191015000270        Y  00105271      케이씨씨             NaN  NaN   
1  20191015000270        Y  00105271      케이씨씨             NaN  NaN   
2  20191015000270        Y  00105271      케이씨씨             NaN  NaN   
3  20191015000270        Y  00105271      케이씨씨             NaN  NaN   

            bddd ctrd gmtsck_shddstd      ap_gmtsck  ...    slprc  \
0  2019년 07월 11일    -  2019년 10월 01일  2019년 11월 13일  ...      NaN   
0            NaN  NaN            NaN            NaN  ...  108,038   
0            NaN  NaN            NaN            NaN  ...      NaN   
1            NaN  NaN            NaN            NaN  ...      NaN   
2            NaN  NaN            NaN            NaN  ...      NaN   
3            NaN  NaN            NaN            NaN  ...      NaN   

              slta    kndn        cnt    cmpnm      sen               tast  \
0              NaN     NaN        NaN      NaN      NaN                NaN   
0  902,143,056,892  기명식보통주  8,350,210      NaN      NaN                NaN   
0              NaN     NaN        NaN  (주)케이씨씨  분할되는 회사  8,805,709,000,000   
1              NaN     NaN        NaN  (주)케이씨씨  분할되는 회사  8,805,709,000,000   
2              NaN     NaN        NaN        -  분할되는 회사                  -   
3              NaN     NaN        NaN        -  분할되는 회사                  -   

              cpt isstk_knd   isstk_cnt  
0             NaN       NaN         NaN  
0             NaN       NaN         NaN  
0  56,435,000,000       보통주  10,556,513  
1  56,435,000,000       우선주           -  
2               -       보통주           -  
3               -       우선주           -  

[6 rows x 29 columns]

### 7. 확장 기능   ###
Open DART의 API가 제공하지 않는 유용한 확장 기능을 제공합니다.

In [29]:
# 지정한 날짜의 공시목록 전체 (시간 정보 포함)

dart.list_date_ex('2022-01-03')

rcept_dt corp_cls      corp_name        rcept_no  \
0   2022-01-03 18:58:00        기            테라셈  20220103900690   
1   2022-01-03 18:57:00        기            테라셈  20220103900691   
2   2022-01-03 18:52:00        코        케이아이엔엑스  20220103900694   
3   2022-01-03 18:37:00        코            에코캡  20220103900693   
4   2022-01-03 18:16:00        코           유아이디  20220103900689   
..                  ...      ...            ...             ...   
497 2022-01-03 07:30:00        기       한국투자신탁운용  20211231100015   
498 2022-01-03 07:30:00        기       한국투자신탁운용  20211231100014   
499 2022-01-03 07:30:00        기  디비씨앤에스자동차손해사정  20211231000496   
500 2022-01-03 07:30:00        기  디비씨앤에스자동차손해사정  20211231000498   
501 2022-01-03 07:30:00        기  디비씨앤에스자동차손해사정  20211231000497   

                                       report_nm         flr_nm  rm  
0                 소송등의판결ㆍ결정(일정금액이상의청구) (부동산인도명령)            테라셈   코  
1                  투자판단관련주요경영사항 (부동산 강제 경매결정 관련)            테라셈  코정  
2                                    단일판매ㆍ공급계약체결        케이아이엔엑스   코  
3                                 타법인주식및출자증권취득결정            에코캡  코정  
4                    유상증자결정(종속회사의주요경영사항) (제3자배정)           유아이디   코  
..                                           ...            ...  ..  
497  효력발생안내 ( 2021.12.30. 제출 일괄신고서(집합투자증권-신탁형) )          금융감독원      
498  효력발생안내 ( 2021.12.30. 제출 일괄신고서(집합투자증권-신탁형) )          금융감독원      
499                                  특수관계인과의내부거래  디비씨앤에스자동차손해사정   공  
500                               특수관계인으로부터부동산임차  디비씨앤에스자동차손해사정   공  
501                               특수관계인으로부터부동산임차  디비씨앤에스자동차손해사정   공  

[502 rows x 7 columns]

보고서의 개별 문서(sub_docs)와 첨부 문서(attach_docs), 첨부 파일(attach_files)에 대한 목록을 가져옵니다.

In [30]:
rcp_no = '20220308000798' # 삼성전자 2021년 사업보고서

In [31]:
# 개별 문서 제목과 URL
dart.sub_docs(rcp_no)

title  \
0                     사 업 보 고 서   
1                【 대표이사 등의 확인 】   
2                     I. 회사의 개요   
3                     1. 회사의 개요   
4                     2. 회사의 연혁   
5                   3. 자본금 변동사항   
6                   4. 주식의 총수 등   
7                  5. 정관에 관한 사항   
8                    II. 사업의 내용   
9                     1. 사업의 개요   
10               2. 주요 제품 및 서비스   
11                3. 원재료 및 생산설비   
12                 4. 매출 및 수주상황   
13               5. 위험관리 및 파생거래   
14             6. 주요계약 및 연구개발활동   
15                   7. 기타 참고사항   
16               III. 재무에 관한 사항   
17                    1. 요약재무정보   
18                    2. 연결재무제표   
19                 3. 연결재무제표 주석   
20                      4. 재무제표   
21                   5. 재무제표 주석   
22                 6. 배당에 관한 사항   
23    7. 증권의 발행을 통한 자금조달에 관한 사항   
24              8. 기타 재무에 관한 사항   
25          IV. 이사의 경영진단 및 분석의견   
26             V. 회계감사인의 감사의견 등   
27               1. 외부감사에 관한 사항   
28               2. 내부통제에 관한 사항   
29      VI. 이사회 등 회사의 기관에 관한 사항   
30                1. 이사회에 관한 사항   
31               2. 감사제도에 관한 사항   
32             3. 주주총회 등에 관한 사항   
33               VII. 주주에 관한 사항   
34       VIII. 임원 및 직원 등에 관한 사항   
35             1. 임원 및 직원 등의 현황   
36                  2. 임원의 보수 등   
37            IX. 계열회사 등에 관한 사항   
38              X. 대주주 등과의 거래내용   
39  XI. 그 밖에 투자자 보호를 위하여 필요한 사항   
40            1. 공시내용 진행 및 변경사항   
41             2. 우발부채 등에 관한 사항   
42              3. 제재 등과 관련된 사항   
43  4. 작성기준일 이후 발생한 주요사항 등 기타사항   
44                     XII. 상세표   
45          1. 연결대상 종속회사 현황(상세)   
46               2. 계열회사 현황(상세)   
47              3. 타법인출자 현황(상세)   
48                4. 연구개발실적(상세)   
49                  【 전문가의 확인 】   
50                   1. 전문가의 확인   
51                2. 전문가와의 이해관계   

                                                  url  
0   http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
1   http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
2   http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
3   http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
4   http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
5   http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
6   http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
7   http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
8   http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
9   http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
10  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
11  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
12  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
13  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
14  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
15  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
16  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
17  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
18  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
19  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
20  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
21  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
22  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
23  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
24  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
25  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
26  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
27  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
28  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
29  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
30  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
31  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
32  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
33  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
34  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
35  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
36  http://dart.fss.or.kr/report/viewer.do?rcpNo=2...  
37  http://dart.fss.or.kr/report/viewer.do

In [32]:
for ix, row in dart.sub_docs(rcp_no).iterrows():
    print(row['title'])
    print(row['url'])

사 업 보 고 서
http://dart.fss.or.kr/report/viewer.do?rcpNo=20220308000798&dcmNo=8446647&eleId=1&offset=797&length=4151&dtd=dart3.xsd
【 대표이사 등의 확인 】
http://dart.fss.or.kr/report/viewer.do?rcpNo=20220308000798&dcmNo=8446647&eleId=2&offset=32912&length=405&dtd=dart3.xsd
I. 회사의 개요
http://dart.fss.or.kr/report/viewer.do?rcpNo=20220308000798&dcmNo=8446647&eleId=3&offset=33321&length=135641&dtd=dart3.xsd
1. 회사의 개요
http://dart.fss.or.kr/report/viewer.do?rcpNo=20220308000798&dcmNo=8446647&eleId=4&offset=33420&length=36187&dtd=dart3.xsd
2. 회사의 연혁
http://dart.fss.or.kr/report/viewer.do?rcpNo=20220308000798&dcmNo=8446647&eleId=5&offset=69611&length=32055&dtd=dart3.xsd
3. 자본금 변동사항
http://dart.fss.or.kr/report/viewer.do?rcpNo=20220308000798&dcmNo=8446647&eleId=6&offset=101670&length=2915&dtd=dart3.xsd
4. 주식의 총수 등
http://dart.fss.or.kr/report/viewer.do?rcpNo=20220308000798&dcmNo=8446647&eleId=7&offset=104589&length=63112&dtd=dart3.xsd
5. 정관에 관한 사항
http://dart.fss.or.kr/report/viewer.do?rcpNo=202203080007

In [33]:
# 첨부 문서 제목과 URL
for ix, row in dart.attach_docs(rcp_no).iterrows():
    print(row['title'])
    print(row['url'])

2022.03.08 감사보고서
http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20220308000798&dcmNo=8446645
2022.03.08 감사의감사보고서
http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20220308000798&dcmNo=8446650
2022.03.08 내부감시장치에대한감사의의견서
http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20220308000798&dcmNo=8446652
2022.03.08 내부회계관리제도운영보고서
http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20220308000798&dcmNo=8446653
2022.03.08 연결감사보고서
http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20220308000798&dcmNo=8446646
2022.03.08 영업보고서
http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20220308000798&dcmNo=8446651
2022.03.08 정관
http://dart.fss.or.kr/dsaf001/main.do?rcpNo=20220308000798&dcmNo=8446649


In [34]:
# 첨부 파일 제목과 URL
files = dart.attach_files(rcp_no)
files

{'[삼성전자]사업보고서(2022.03.08).pdf': 'http://dart.fss.or.kr/pdf/download/pdf.do?rcp_no=20220308000798&dcm_no=8446647',
 '[삼성전자]사업보고서_재무제표(2022.03.08)_ko.xls': 'http://dart.fss.or.kr/pdf/download/excel.do?rcp_no=20220308000798&dcm_no=8446647&lang=ko',
 '[삼성전자]사업보고서_IFRS(원문XBRL)(2022.03.08).zip': 'http://dart.fss.or.kr/pdf/download/ifrs.do?rcp_no=20220308000798&dcm_no=8446647&lang=ko'}

In [35]:
# 첨부 파일 모두 다운로드 하기

files = dart.attach_files(rcp_no)
for title, url in files.items():
    print(title)
    print(url)
    dart.download(url, title)

[삼성전자]사업보고서(2022.03.08).pdf
http://dart.fss.or.kr/pdf/download/pdf.do?rcp_no=20220308000798&dcm_no=8446647
[삼성전자]사업보고서_재무제표(2022.03.08)_ko.xls
http://dart.fss.or.kr/pdf/download/excel.do?rcp_no=20220308000798&dcm_no=8446647&lang=ko
[삼성전자]사업보고서_IFRS(원문XBRL)(2022.03.08).zip
http://dart.fss.or.kr/pdf/download/ifrs.do?rcp_no=20220308000798&dcm_no=8446647&lang=ko


In [36]:
ls "[삼성전자]*"

 G 드라이브의 볼륨: Google Drive
 볼륨 일련 번호: 1983-1116

 G:\내 드라이브\60 FD - 전자공시\900 OpenDartReader\OpenDartReader\docs 디렉터리

2022-10-20  오전 01:41         2,038,716 [삼성전자]사업보고서(2022.03.08).pdf
2022-10-20  오전 01:41           216,730 [삼성전자]사업보고서_IFRS(원문XBRL)(2022.03.08).zip
2022-10-20  오전 01:41            53,248 [삼성전자]사업보고서_재무제표(2022.03.08)_ko.xls
               3개 파일           2,308,694 바이트
               0개 디렉터리  698,479,546,368 바이트 남음


In [37]:
import pandas as pd

xl = pd.ExcelFile('[삼성전자]사업보고서_재무제표(2022.03.08)_ko.xls')
xl.sheet_names

['기본정보',
 '연결 재무상태표',
 '연결 손익계산서',
 '연결 포괄손익계산서',
 '연결 자본변동표',
 '연결 현금흐름표',
 '재무상태표',
 '손익계산서',
 '포괄손익계산서',
 '자본변동표',
 '현금흐름표']

In [38]:
pd.read_excel('[삼성전자]사업보고서_재무제표(2022.03.08)_ko.xls', sheet_name='연결 손익계산서', skiprows=6)

제 53 기     제 52 기     제 51 기
0                          수익(매출액)  279604799  236806988  230400881
1                             매출원가  166411342  144488296  147239549
2                            매출총이익  113193457   92318692   83161332
3                          판매비와관리비   61559601   56324816   55392823
4                             영업이익   51633856   35993876   27768509
5                             기타수익    2205695    1384068    1778666
6                             기타비용    2055971    2488902    1414707
7                            지분법이익     729614     506530     412960
8                             금융수익    8543187   12267600   10161632
9                             금융비용    7704554   11318055    8274871
10                 법인세비용차감전순이익(손실)   53351827   36345117   30432189
11                           법인세비용   13444377    9937285    8693324
12                      계속영업이익(손실)   39907450   26407832   21738865
13                       당기순이익(손실)   39907450   26407832   21738865
14                   당기순이익(손실)의 귀속                                 
15      지배기업의 소유주에게 귀속되는 당기순이익(손실)   39243791   26090846   21505054
16           비지배지분에 귀속되는 당기순이익(손실)     663659     316986     233811
17                            주당이익                                 
18             기본주당이익(손실) (단위 : 원)       5777       3841       3166
19             희석주당이익(손실) (단위 : 원)       5777       3841       3166

***2020-2022  [FinanceData.KR]() | [facebook.com/financedata]()***